nifi-extension-bundles/nifi-snowflake-bundle/nifi-snowflake-services/src/main/java/org/apache/nifi/snowflake/service/SnowflakeComputingConnectionPool.java

```diff
@@ -280,6 +311,7 @@ protected Map<String, String> getConnectionProperties(final ConfigurationContext
 
     @Override
     public SnowflakeConnectionWrapper getSnowflakeConnection() {
+        refreshAccessToken();
```

Comment:
`getSnowflakeConnection()` interface method is specific to the `PutSnowflakeInternalStage` processor. The controller service can also be used from the generic SQL processors (like `QueryDatabaseTableRecord`) and they simply call `getConnection()`. For this reason, the access token must be refreshed in that method.